apiserver/facades/agent/uniter/uniter_network_test.go

```diff
@@ -770,6 +773,20 @@ func (s *uniterNetworkInfoSuite) TestCommitHookChanges(c *gc.C) {
 	c.Assert(unitPortRanges.UniquePortRanges(), jc.DeepEquals, []network.PortRange{{Protocol: "tcp", FromPort: 80, ToPort: 81}})
 	c.Assert(unitPortRanges.ByEndpoint()["monitoring-port"], jc.DeepEquals, []network.PortRange{{Protocol: "tcp", FromPort: 80, ToPort: 81}}, gc.Commentf("unit ports where not opened for the requested endpoint"))
 
+	unitUUID, err := s.domainServices.Application(service.ApplicationServiceParams{}).GetUnitUUID(context.Background(), s.wordpressUnit.Tag().Id())
+	c.Assert(err, jc.ErrorIsNil)
+	grp, err := s.domainServices.Port().GetUnitOpenedPorts(context.Background(), unitUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(grp, jc.DeepEquals, network.GroupedPortRanges{
+		"monitoring-port": {
+			{Protocol: "tcp", FromPort: 80, ToPort: 81},
+			// NOTE: Opening and closing the same port range at the same time
+			// leads to different behaviour between DQLite and Mongo
+			// TODO (jack-w-shaw): evaluate if this is worth fixing
+			{Protocol: "tcp", FromPort: 7337, ToPort: 7337},
```

Comment:
There is a pending task to re-evaluate the ports domain, around RunAtomic and that

I have left a comment here to remind me about this specific 'issue'

https://github.com/juju/juju/pull/18147#issuecomment-2393542463